langchain/utilities/bibtex.py

```diff
@@ -0,0 +1,177 @@
+                "Please install it with `pip install bibtexparser`."
+            )
+        return values
+
+    def _meta_str(self, entry: Any) -> str:
+        """
+        Return the meta information of an article in string format.
+        """
+        metadata = self._get_meta(entry)
+        return "\n".join([f"{key}: {value}" for key, value in metadata.items()])
+
+    def _get_metata(self, entry: Any) -> str:
+        meta = {
+            "ID": entry.get("ID", ""),
+            "Published": entry.get("year", "Published date unknown"),
+            "Title": entry.get("title", "Unknown"),
+            "Publication": entry.get("journal")
+            or entry.get("booktitle")
+            or "Unknown publication",
+            "Authors": entry.get("author", "Unknown authors"),
+            "Summary": entry.get("abstract", "No abstract available"),
+            "URL": entry.get("url", None) or f'https://doi.org/{entry.get("doi", "")}'
+            if entry.get("doi", None)
+            else "No URL available",
+        }
+        optional_fields = [
+            "annote",
+            "booktitle",
+            "editor",
+            "howpublished",
+            "journal",
+            "keywords",
+            "note",
+            "organization",
+            "publisher",
+            "school",
+            "series",
+            "type",
+            "doi",
+            "issn",
+            "isbn",
+        ]
+        if self.load_all_available_meta:
+            for field in optional_fields:
+                if field in entry:
+                    meta[field.capitalize()] = entry.get(field)
+        return meta
+
+    def run(self, file_path: str) -> str:
+        """
+        Load bibtex file using bibtexparser and get the article meta information.
+        See https://bibtexparser.readthedocs.io/en/master/
+        It uses only the most informative fields of article meta information.
+        """
+        print("Running bibtexparser wrapper")
+        try:
+            with open(file_path) as file:
+                docs = [
+                    self._meta_str(entry)
+                    for entry in self.load_file(file).entries[: self.load_max_docs]
+                ]
+            return (
+                "\n\n".join(docs)[: self.doc_content_chars_max]
+                if docs
+                else "No good bibtex information found. Check your bibtex file."
+            )
+        except self.bibtexparser_exceptions as ex:
+            return f"Bibtexparser exception: {ex}"
+
+    def load(self, file_path: str) -> List[Document]:
+        """
+        Load bibtex file using bibtexparser and get the article texts plus the
+        article metadata.
+        See https://bibtexparser.readthedocs.io/en/master/
+
+        Returns: a list of documents with the document.page_content in text format
+
+        """
+        print("Running bibtexparser wrapper LOADER")
+        try:
+            import fitz
```

Comment:
We'll need to swap out the default parser implementation here so fitz is not hard-coded